dbt/adapters/databricks/python_submissions.py

```diff
@@ -1,75 +1,65 @@
-import os
-from typing import Dict, cast
-import uuid
+from typing import Dict, Optional
 
-from requests import HTTPError
-
-import dbt.exceptions
-from dbt.adapters.spark.python_submissions import BaseDatabricksHelper
+from dbt.adapters.spark.python_submissions import (
+    AllPurposeClusterPythonJobHelper,
+    BaseDatabricksHelper,
+    JobClusterPythonJobHelper,
+)
 
 from dbt.adapters.databricks.__version__ import version
-from dbt.adapters.databricks.api_client import Api12Client
-from dbt.adapters.databricks.connections import DatabricksCredentials, DBT_DATABRICKS_INVOCATION_ENV
+from dbt.adapters.databricks.connections import DatabricksCredentials
 
 
-class CommandApiPythonJobHelper(BaseDatabricksHelper):
+class DbtDatabricksBasePythonJobHelper(BaseDatabricksHelper):
     credentials: DatabricksCredentials  # type: ignore[assignment]
 
     def __init__(self, parsed_model: Dict, credentials: DatabricksCredentials) -> None:
         super().__init__(
             parsed_model=parsed_model, credentials=credentials  # type: ignore[arg-type]
         )
 
-        command_name = f"dbt-databricks_{version}"
-        invocation_env = os.environ.get(DBT_DATABRICKS_INVOCATION_ENV)
-        if invocation_env is not None and len(invocation_env) > 0:
-            command_name = f"{command_name}-{invocation_env}"
-        command_name += "-" + str(uuid.uuid1())
+        user_agent = f"dbt-databricks/{version}"
+
+        invocation_env = credentials.get_invocation_env()
+        if invocation_env:
+            user_agent = f"{user_agent} ({invocation_env})"
+
+        connection_parameters = credentials.connection_parameters.copy()  # type: ignore[union-attr]
 
-        self.api_client = Api12Client(
-            host=cast(str, credentials.host),
-            token=cast(str, credentials.token),
-            command_name=command_name,
+        http_headers: Dict[str, str] = credentials.get_all_http_headers(
+            connection_parameters.pop("http_headers", {})
         )
 
-    def check_credentials(self) -> None:
-        if not self.cluster_id:
-            raise ValueError("Databricks cluster is required for commands submission method.")
+        self.auth_header.update({"User-Agent": user_agent, **http_headers})
 
-    def submit(self, compiled_code: str) -> None:
-        cluster_id = self.cluster_id
+    @property
+    def cluster_id(self) -> Optional[str]:  # type: ignore[override]
+        return self.parsed_model.get(
+            "cluster_id",
+            self.credentials.extract_cluster_id(
+                self.parsed_model.get("http_path", self.credentials.http_path)
```

Comment:
Can we specify an `http_path` that is different from the one in dbt profiles for a Python model?

Reply:
Thanks for the heads-up.
Actually I double-checked it and found there is a mistake. I'll update it.

After the update, yes, we can specify `http_path` in the Python model config that is different from the one in the profile.